qiskit-superstaq/qiskit_superstaq/superstaq_backend.py

```diff
@@ -210,20 +210,22 @@ def aqt_compile(
             ValueError: If this is not an AQT backend.
         """
         if not self.name().startswith("aqt_"):
-            raise ValueError(f"{self.name()} is not a valid AQT target.")
+            raise ValueError(f"{self.name()!r} is not a valid AQT target.")
 
         options: Dict[str, Any] = {**kwargs}
         if num_equivalent_circuits is not None:
-            options["num_equivalent_circuits"] = num_equivalent_circuits
+            gss.validation.validate_integer_param(num_equivalent_circuits)
+            options["num_equivalent_circuits"] = int(num_equivalent_circuits)
         if random_seed is not None:
-            options["random_seed"] = random_seed
+            gss.validation.validate_integer_param(random_seed)
+            options["random_seed"] = int(random_seed)
         if atol is not None:
-            options["atol"] = atol
+            options["atol"] = float(atol)
         if gate_defs is not None:
             options["gate_defs"] = gate_defs
 
```

Comment:
Check if options is None before converting to json here?

Reply:
done (updated in `_get_compile_request_json()`)